src/unistd.rs

```diff
@@ -1324,6 +1324,79 @@ pub fn pause() {
     unsafe { libc::pause() };
 }
 
+pub mod alarm {
+    //! Alarm signal scheduling.
+    //!
+    //! See also [alarm](http://pubs.opengroup.org/onlinepubs/9699919799/functions/alarm.html).
+    //!
+    //! # Notes
+    //!
+    //! Scheduling an alarm will trigger a `SIGALRM` signal when the the time
+    //! has elapsed, which has to be caught, because the default action for the
+    //! signal is to terminate the program. This signal also can't be ignored
+    //! because the system calls like `pause` will not be interrupted, see the
+    //! second example below.
+    //!
+    //! # Examples
+    //!
+    //! Canceling an alarm.
+    //!
+    //! ```
+    //! use nix::unistd::alarm;
+    //!
+    //! // Set an alarm for 60 seconds from now.
+    //! alarm::set(60);
+    //!
+    //! // Cancel the above set alarm, which returns the number of seconds left
+    //! // of the previously set alarm.
+    //! assert_eq!(alarm::cancel(), 60);
+    //! ```
+    //!
+    //! Scheduling an alarm and waiting for the signal.
+    //!
+    //! ```
+    //! use std::time::{Duration, Instant};
+    //!
+    //! use nix::unistd::{alarm, pause};
+    //! use nix::sys::signal::*;
+    //!
+    //! // We need to setup an empty signal handler to catch the alarm signal,
+    //! // otherwise the program will be terminated once the signal is delivered.
+    //! extern fn signal_handler(_: nix::libc::c_int) { }
+    //! unsafe { sigaction(Signal::SIGALRM, &SigAction::new(SigHandler::Handler(signal_handler), SaFlags::empty(), SigSet::empty())); }
+    //!
+    //! // Set an alarm for 1 second from now.
+    //! alarm::set(1);
+    //!
+    //! let start = Instant::now();
+    //! // Pause the process until the alarm signal is received.
+    //! pause();
+    //!
+    //! assert!(start.elapsed() >= Duration::from_secs(1));
+    //! ```
+
+    use libc;
+
+    /// Schedule an alarm signal.
+    ///
+    /// This will cause the system to generate a `SIGALRM` signal for the
+    /// process after the number of `secs` have elapsed.
+    ///
+    /// Returns the leftover time of a previously set alarm, or `0` if no alarm
+    /// was set.
+    pub fn set(secs: libc::c_uint) -> libc::c_uint {
```

Comment:
Should we check for 0 in the input argument with an assertion? I think if people are doing that they should really be using `cancel` instead of `set`.

Reply:
I could add a simple assertion.

Reply:
What about this:
```
assert!(secs != 0, "passing 0 to `alarm::set` is not allowed, to cancel an alarm use `alarm::cancel`");
```

Reply:
Sure, that'd be fine.